constants.php

```diff
@@ -1,35 +1,64 @@
 <?php
+//NB: Do not edit; use ./constants.local.php instead.
+
+//<Not customisable>
 define('FRESHRSS_VERSION', '1.8.1-dev');
 define('FRESHRSS_WEBSITE', 'https://freshrss.org');
 define('FRESHRSS_WIKI', 'https://freshrss.github.io/FreshRSS/');
 
-define('FRESHRSS_USERAGENT', 'FreshRSS/' . FRESHRSS_VERSION . ' (' . PHP_OS . '; ' . FRESHRSS_WEBSITE . ')');
-
-// PHP text output compression http://php.net/ob_gzhandler (better to do it at Web server level)
-define('PHP_COMPRESSION', false);
-
-// Maximum log file size in Bytes, before it will be divided by two
-define('MAX_LOG_SIZE', 1048576);
+define('FRESHRSS_PATH', __DIR__);
+define('PUBLIC_PATH', FRESHRSS_PATH . '/p');
+define('PUBLIC_TO_INDEX_PATH', '/i');
+define('INDEX_PATH', PUBLIC_PATH . PUBLIC_TO_INDEX_PATH);
+define('PUBLIC_RELATIVE', '..');
+define('LIB_PATH', FRESHRSS_PATH . '/lib');
+define('APP_PATH', FRESHRSS_PATH . '/app');
+define('EXTENSIONS_PATH', FRESHRSS_PATH . '/extensions');
+//</Not customisable>
 
-// Constantes de chemins
-define('FRESHRSS_PATH', dirname(__FILE__));
-
-	define('PUBLIC_PATH', FRESHRSS_PATH . '/p');
-		define('PUBLIC_TO_INDEX_PATH', '/i');
-		define('INDEX_PATH', PUBLIC_PATH . PUBLIC_TO_INDEX_PATH);
-		define('PUBLIC_RELATIVE', '..');
+if (file_exists(__DIR__ . '/constants.local.php')) {
+	//Include custom / local settings:
+	include(__DIR__ . '/constants.local.php');
+}
 
+if (!defined('FRESHRSS_USERAGENT')) {
```

Comment:
Do you think this whole block would be more elegant with some kind of array/for loop construction?

Reply:
Could you please explain a bit more what you have in mind?

Reply:
I could propose something like:

```php
function safe_define($name, $value) {
	if (!defined($name) {
		return define($name, $value);
	}
}

safe_define('MAX_LOG_SIZE', 1048576);
safe_define('DATA_PATH', FRESHRSS_PATH . '/data');
...
```

Reply:
Yes, that's fine too. I'm not saying I object per se, although I'm not overly fond of how this looks, just that I'd like it to be a conscious choice. :-)

I meant simply something like

```php
$defines = [
    "MAX_LOG_SIZE" => 1234,
    DATA_PATH => '/data',
];

foreach ($defines as $key => $value) {
  safe_define($key, $value)
}
```

Reply:
Let's go for it, then :-) https://github.com/FreshRSS/FreshRSS/pull/1725/commits/5dd2906eb2e9a19a1f3277880bc37a259d3cb7ec